pkg/log/logshipper/logshipper.go

```diff
@@ -0,0 +1,127 @@
+package logshipper
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/kolide/launcher/pkg/agent/types"
+	"github.com/kolide/launcher/pkg/sendbuffer"
+)
+
+// TODO: consolidate this var, but where?
+var observabilityIngestTokenKey = []byte("observability_ingest_auth_token")
+
+const (
+	truncatedFormatString = "%s[TRUNCATED]"
+)
+
+type LogShipper struct {
+	sender            *authedHttpSender
+	sendBuffer        *sendbuffer.SendBuffer
+	logger            log.Logger
+	knapsack          types.Knapsack
+	stopFunc          context.CancelFunc
+	isShippingEnabled bool
+}
+
+func New(k types.Knapsack) (*LogShipper, error) {
+	token, _ := k.TokenStore().Get(observabilityIngestTokenKey)
+
+	logEndpoint, err := logEndpoint(k)
+	if err != nil {
+		return nil, err
+	}
+
+	sender := newAuthHttpSender(logEndpoint, string(token))
+
+	sendInterval := time.Minute * 1
+	if k.Debug() {
+		sendInterval = time.Second * 1
+	}
+
+	sendBuffer := sendbuffer.New(sender, sendbuffer.WithSendInterval(sendInterval))
+	logger := log.NewJSONLogger(sendBuffer)
+
+	return &LogShipper{
+		sender:     sender,
+		sendBuffer: sendBuffer,
+		logger:     logger,
+		knapsack:   k,
```

Comment:
I think you'd want to set `isShippingEnabled: ls.knapsack.LogIngestServerURL() != ""` here now too, otherwise you'd have to wait for a subscriber update to start shipping logs.

Reply:
nice catch!